cubical/machines/interval_gain_machine.py

```diff
@@ -559,12 +559,18 @@ def issue_warning(msg, sort_index=70):
                 if log.verbosity() > 1:
                     msg = "{} directions ({}) flagged {}".format(
                         len(bad_dirs),
-                        ", ".join(["dir {0:s}: {1:.3f}% gains affected".format(
-                            str(d), percflagged[d]) for d in bad_dirs]),
-                        why_flagged
-                    )
+                        ", ".join(["dir {0:s}: {1:s} gains affected".format(
```

Comment:
I would suggest constructing the message incrementally in an if/else ladder. This is very opaque (to me). Disregard if you feel strongly.

Reply:
Sorry. As you can tell I like golfing...